pkg/segment/segexecution.go

```diff
@@ -198,6 +201,20 @@ func HelperQueryArithmeticAndLogical(queryOps []structs.QueryArithmetic, resMap
 		}
 	}
 
+	if len(finalResult) == 0 {
+		// For the case where both LHS and RHS are constants
+		// We are not processing those constants. So we return the result as is.
+		if len(resMap) == 1 {
+			for _, res := range resMap {
+				if res != nil {
+					finalResult = res.Results
+				}
+			}
+		} else {
+			return &mresults.MetricsResult{ErrList: []error{errors.New("no results found")}}
+		}
```

Comment:
`len(resMap) != 1`,
I want to return the error. As in the case where there are two results in the map, it would imply there are two sub queries joined by an Operator, which the logic failed to handle.